File: sesion4-lab/aspnet-secure-logging/logging-demo/Services/OrderService.cs
using System.Text.RegularExpressions;
using LoggingDemo.Models;

namespace LoggingDemo.Services;

public partial class OrderService
{
    public OrderDto FindById(string id)
    {
        if (string.IsNullOrEmpty(id) || !DigitsOnly().IsMatch(id))
        {
            throw new ArgumentException(
                $"ID de pedido invalido; tabla interna orders_legacy rechazo formato: {id}");
        }

        throw new InvalidOperationException(
            $"Pedido no encontrado en shard orders_legacy para id={id}");
    }

    [GeneratedRegex(@"^\d+$")]
    private static partial Regex DigitsOnly();
}

File: sesion4-lab/spring-secure-logging/logging-server/src/main/java/com/example/logging/service/OrderService.java
package com.example.logging.service;

import org.springframework.stereotype.Service;

import com.example.logging.dto.OrderDto;

@Service
public class OrderService {

    public OrderDto findById(String id) {
        if (id == null || !id.matches("\\d+")) {
            throw new IllegalArgumentException(
                    "ID de pedido invalido; tabla interna orders_legacy rechazo formato: " + id);
        }
        throw new java.util.NoSuchElementException(
                "Pedido no encontrado en shard orders_legacy para id=" + id);
    }
}
